### PR TITLE
Fix server error in RSS feeds with unexpected judgment dates

### PR DESCRIPTION
**Pull request by jacksonj04**
New behaviour in `SearchResult` introduced in https://github.com/nationalarchives/ds-caselaw-public-ui/pull/652 would result in cases where `date` was a string, rather than a `datetime`. This behaviour is changed so that it returns `None`.

A string would cause the RSS library to fail; a `None` is correctly handled.

### CODE REVIEW

File: judgments/models.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def __init__(
         try:
             self.date = dateparser.parse(date)
         except ParserError:
-            self.date = date
+            self.date = None
```
Comment (dragon-dxw) [inline judgments/models.py:33]: It might be worth flagging a rollbar error if the date is not empty and fails to parse; but it shouldn't raise.

**Review by dragon-dxw — APPROVED** [review]
Whilst there's improvements that can be made, this is a production problem, and this feels like a good fix.

File: judgments/tests/test_atom_feed.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @patch("judgments.utils.perform_advanced_search")
+    def test_bad_page_404(self, fake_advanced_search):
+        # "?page=" 404s, not 500
+        fake_advanced_search.return_value = fake_search_results()
+        response = self.client.get("/atom.xml?page=")
+        self.assertEqual(response.status_code, 404)
+
+    @patch("judgments.feeds.perform_advanced_search")
+    @patch("judgments.models.SearchResult.create_from_node")
+    def test_feed_with_empty_date(self, fake_result, fake_advanced_search):
```
[the remark this answers is not in the document]
Comment (jacksonj04) [reply judgments/tests/test_atom_feed.py:36]: Backported some testing for this from the refactor work, just for belt and braces.

File: judgments/models.py (hunk 1 above)
Comment (jacksonj04) [reply judgments/models.py:38]: Added with tests

**Pull request merged**